src/id3/v2/tag.rs

```diff
@@ -1634,4 +1734,278 @@ mod tests {
+
+		assert_eq!(tag.track().unwrap(), track_number);
+		assert_eq!(tag.track_total().unwrap(), track_total);
+	}
+
+	#[test]
+	fn disk_number_tag_to_id3v2() {
+		use crate::traits::Accessor;
+		let disk_number = 1;
+
+		let mut tag = Tag::new(TagType::ID3v2);
+
+		tag.push_item(TagItem::new(
+			ItemKey::DiscNumber,
+			ItemValue::Text(disk_number.to_string()),
+		));
+
+		let tag: ID3v2Tag = tag.into();
+
+		assert_eq!(tag.disk().unwrap(), disk_number);
+		assert!(tag.disk_total().is_none());
+	}
+
+	#[test]
+	fn disk_total_tag_to_id3v2() {
+		use crate::traits::Accessor;
+		let disk_total = 2;
+
+		let mut tag = Tag::new(TagType::ID3v2);
+
+		tag.push_item(TagItem::new(
+			ItemKey::DiscTotal,
+			ItemValue::Text(disk_total.to_string()),
+		));
+
+		let tag: ID3v2Tag = tag.into();
+
+		assert_eq!(tag.disk().unwrap(), DEFAULT_NUMBER_IN_PAIR);
+		assert_eq!(tag.disk_total().unwrap(), disk_total);
+	}
+
+	#[test]
+	fn disk_number_and_disk_total_tag_to_id3v2() {
+		use crate::traits::Accessor;
+		let disk_number = 1;
+		let disk_total = 2;
+
+		let mut tag = Tag::new(TagType::ID3v2);
+
+		tag.push_item(TagItem::new(
+			ItemKey::DiscNumber,
+			ItemValue::Text(disk_number.to_string()),
+		));
+
+		tag.push_item(TagItem::new(
+			ItemKey::DiscTotal,
+			ItemValue::Text(disk_total.to_string()),
+		));
+
+		let tag: ID3v2Tag = tag.into();
+
+		assert_eq!(tag.disk().unwrap(), disk_number);
+		assert_eq!(tag.disk_total().unwrap(), disk_total);
+	}
+
+	fn create_tag_with_trck_and_tpos_frame(content: &'static str) -> Tag {
+		fn insert_frame(id: &'static str, content: &'static str, tag: &mut ID3v2Tag) {
+			tag.insert(new_text_frame(
+				FrameID::Valid(Cow::Borrowed(id)),
+				content.to_string(),
+				FrameFlags::default(),
+			));
+		}
+
+		let mut tag = ID3v2Tag::default();
+
+		insert_frame("TRCK", content, &mut tag);
+		insert_frame("TPOS", content, &mut tag);
+
+		tag.into()
+	}
+
+	#[test]
+	fn valid_trck_and_tpos_frame() {
+		fn assert_valid(content: &'static str, number: Option<u32>, total: Option<u32>) {
+			let tag = create_tag_with_trck_and_tpos_frame(content);
+
+			assert_eq!(tag.track(), number);
+			assert_eq!(tag.track_total(), total);
+			assert_eq!(tag.disk(), number);
+			assert_eq!(tag.disk_total(), total);
+		}
+
+		assert_valid("0", Some(0), None);
+		assert_valid("1", Some(1), None);
+		assert_valid("0/0", Some(0), Some(0));
+		assert_valid("1/2", Some(1), Some(2));
+		assert_valid("010/011", Some(10), Some(11));
+	}
+
+	#[test]
+	fn invalid_trck_and_tpos_frame() {
+		fn assert_invalid(content: &'static str) {
+			let tag = create_tag_with_trck_and_tpos_frame(content);
+
+			assert!(tag.track().is_none());
+			assert!(tag.track_total().is_none());
+			assert!(tag.disk().is_none());
+			assert!(tag.disk_total().is_none());
+		}
+
+		assert_invalid("");
+		assert_invalid(" ");
+		assert_invalid("/");
+		assert_invalid("/1");
+		assert_invalid("1/");
+		assert_invalid("a/b");
+		assert_invalid("1 / 2");
```

Comment:
I would expect that leading/trailing whitespaces around the numbers are trimmed.

https://en.wikipedia.org/wiki/Robustness_principle